### DISCUSSION TITLE
Como vocês definem o conceito de mobile em código?

### DISCUSSION THREAD

**Discussion by omariosouto** [frontendbr/forum#510] [Perguntas e Respostas]
Recentemente estive em um projeto em que precisei criar diversas interações que existem somente em dispositivos Desktop e algumas delas precisam ser adaptadas para Tablets e outras só para o Celular.

Compactuo bastante com a ideia do @afonsopacifer no vídeo https://www.youtube.com/watch?v=Rrys1PVR5tQ&t=1s que a tela não é mais um parâmetro e que devemos oferecer o máximo possível experiências que funcionem independente da plataforma (porém nem sempre esse é o cenário que acontece).

Gostaria de saber de situações que vocês já tiveram com isso, como solucionaram e jogo um exemplo do problema:

Você possui um slider com vídeo, onde esse vídeo não pode aparecer em dispositivos "mobile" e deverá aparecer uma imagem no lugar para economizar consumo de banda do usuário. (Considerando que vai ter que abrir em um iPad Pro sem o vídeo)

**Comment by mksanti** [comment]
Nesse caso acredito que a melhor opção seja detectar o OS/Browser e renderizar esse componente de acordo a situação. O jeito de renderizar vai depender das tecnologias que estiver utilizando tanto.

**Comment by felipefialho** [comment]
No seu caso especifico, pode usar o `window.matchMedia()` para remover o vídeo em telas que sejam menores que tablet (ou pelo menos deixa-lo pausado por default, o usuário ativa se quiser). O que alias, concordo totalmente.

Quanto as resoluções, tenho um range de breakpoints e sigo ele na maioria dos projetos.
